corgi/api/views.py

```diff
@@ -342,20 +342,24 @@ class ProductStreamViewSetSet(ProductDataViewSet):
     )
     def list(self, request: Request, *args: tuple, **kwargs: dict) -> Response:
         req = self.request
+        ps_ofuri = req.query_params.get("ofuri")
+        ps_name = req.query_params.get("name")
         active = request.query_params.get("active")
         if active == "all":
             self.queryset = ProductStream.objects.order_by(super().ordering_field).using(
                 "read_only"
             )
-        ofuri = req.query_params.get("ofuri")
-        if not ofuri:
+        if not ps_ofuri and not ps_name:
             return super().list(request)
-        ps = ProductStream.objects.filter(ofuri=ofuri).using("read_only").first()
-        if not ps:
+        try:
+            if ps_name:
+                ps = ProductStream.objects.get(name=ps_name)
+            elif ps_ofuri:
+                ps = ProductStream.objects.get(ofuri=ps_ofuri)
+            serializer = ProductStreamSerializer(ps)
+            return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
even better ;) will investigate

Reply:
hmm after playing about - this was the best I could come up with (eg. call super.retrieve() and override with get_object) ... otherwise there is no redirect we are performing as we want to preserve the URL ... in this case identity is both a pk (eg. a /{uuid}) and ?name= or ?ofuri= for product stream ... getting all three variants to work with minimal code looks like this for me.

it is likely those who know django/drf better will come up with something cleaner.

Reply:
but we are not proposing a /{ofuri} or /{purl} path in the URL... remember both purl and ofuri are also URIs hence passing via param which is where all the awkwardness starts.

We could entertain /{name} for product streams but thought it best to keep url path just for /{uuid}

as for lookup_field I did try that initially and was not able to get the behaviour that preserves url ... will try again (and thx for the inspiration)

Reply:
URIs passed in a URI can only be unambiguously encoded within a url param eg. cannot safely be embedded in the url path itself .... hence the need for ?ofuri={ofuri} and ?purl={purl} ... ok I am going to play around a bit more.